lib/Backends/CPU/libjit/libjit.cpp

```diff
@@ -2757,4 +2757,275 @@ libjit_nms_i32(int32_t *indices, int32_t *numDetected, const float *boxTensor,
+  dim_t stageNum = std::log2((double)fftLength);
+
+  // Number of radix2 butterfly groups for 1st stage.
+  dim_t groupNum = fftLength / 2;
+
+  // Number of radix2 butterflies per group for 1st stage.
+  dim_t groupButterNum = 1;
+
+  // Stage loop.
+  for (dim_t stageIdx = 0; stageIdx < stageNum; stageIdx++) {
+
+    // Butterfly input/output pointers.
+    float *inp1Ptr = bitRevOut + 0 * groupButterNum;
+    float *inp2Ptr = bitRevOut + 2 * groupButterNum;
+
+    // Butterfly group loop.
+    for (dim_t groupIdx = 0; groupIdx < groupNum; groupIdx++) {
+
+      // Twiddle factors pointer.
+      const float *twPtr = twiddleFactors;
+
+      // Butterfly loop within group.
+      for (dim_t groupButterIdx = 0; groupButterIdx < groupButterNum;
+           groupButterIdx++) {
+
+        // Radix 2 butterfly.
+        float inp0_re = *inp1Ptr++;
+        float inp0_im = *inp1Ptr--;
+        float inp1_re = *inp2Ptr++;
+        float inp1_im = *inp2Ptr--;
+
+        float tw_re = *twPtr++;
+        float tw_im = *twPtr--;
+        twPtr += (2 * groupNum);
+
+        float inp1_tw_mult_re = inp1_re * tw_re - inp1_im * tw_im;
+        float inp1_tw_mult_im = inp1_re * tw_im + inp1_im * tw_re;
+
+        *inp1Ptr++ = inp0_re + inp1_tw_mult_re;
+        *inp1Ptr++ = inp0_im + inp1_tw_mult_im;
+        *inp2Ptr++ = inp0_re - inp1_tw_mult_re;
+        *inp2Ptr++ = inp0_im - inp1_tw_mult_im;
+      }
+
+      inp1Ptr += 2 * groupButterNum;
+      inp2Ptr += 2 * groupButterNum;
+    }
+
+    // Update parameters for next stage.
+    groupNum >>= 1;
+    groupButterNum <<= 1;
+  }
+}
+
+/// FFT Radix2 DIT (Decimation In Time) implementation for Real data.
+/// The implementation uses a fftLength/2 FFT for Complex data followed
+/// by a step to map the complex FFT to the real FFT by using a set of
+/// of complex weights \p complexToRealWeights A[k] defined as:
+///   A[k] = 1/2 * (1 - j*exp(-j*2*pi*k/N)) for k = 0 .. N/4-1
+/// The \p input buffer has \p fftLength float values corresponding
+/// to \p fftLength real samples. Since the FFT of a real signal
+/// has conjugate symmetry, the \p output buffer only contains
+/// 2 * (fftLength/2+1) = fftLength + 2 float values corresponding
+/// to fftLength/2+1 complex samples with real and imaginary parts
+/// interleaved: real[0], imag[0], real[1], imag[1], ...
+/// The lookup tables \p twiddleFactors and \p bitReverseIndices are
+/// generated at compile time as if they were generated for a N/2
+/// complex FFT. The boolean flag \p inPlace decides whether the FFT
+/// computation is done in-place (that is in the \p input buffer
+/// without writing in the \p output buffer) or out-of-place (written in
+/// the \p output buffer).
+void libjit_fft_real_f(float *output, float *input, const float *twiddleFactors,
+                       const int32_t *bitReverseIndices,
+                       const float *complexToRealWeights, unsigned fftLength,
+                       bool inPlace) {
```

Comment:
Is `inPlace` used anywhere? Looks like it's always false, unless it's going to be used in a future PR?

Reply:
It`s not used right now but I put it there since the FFT has the ability to do in-place computation.
Future PR`s might use it.